scanpipe/apps.py

```diff
@@ -20,45 +20,51 @@
 # ScanCode.io is a free software code scanning tool from nexB Inc. and others.
 # Visit https://github.com/nexB/scancode.io for support and download.
 
-from pathlib import Path
-
 from django.apps import AppConfig
+from django.core.exceptions import ImproperlyConfigured
 from django.utils.translation import gettext_lazy as _
 
-dot_py_suffix = ".py"
-
+try:
+    from importlib import metadata as importlib_metadata
+except ImportError:
+    import importlib_metadata
 
-def remove_dot_py_suffix(filename):
-    """
-    Return the `filename` without the trailing ".py" suffix if any.
-    """
-    if filename.endswith(dot_py_suffix):
-        return filename[: -len(dot_py_suffix)]
-    return filename
+from scanpipe.pipelines import is_pipeline_subclass
 
 
 class ScanPipeConfig(AppConfig):
     name = "scanpipe"
     verbose_name = _("ScanPipe")
-    pipelines = []
+    pipelines = {}
 
     def ready(self):
+        self.load_pipelines()
+
+    def load_pipelines(self):
         """
-        Load available Pipelines.
+        Load Pipelines from the "scancodeio_pipelines" entry point group.
         """
-        project_root = Path(__file__).parent.parent.absolute()
-        pipelines_dir = project_root / "scanpipe" / "pipelines"
+        entry_points = importlib_metadata.entry_points()
 
-        for child in pipelines_dir.iterdir():
-            if child.name.endswith(dot_py_suffix) and not child.name.startswith("_"):
-                location = str(child.relative_to(project_root))
-                name = remove_dot_py_suffix(child.name)
-                self.pipelines.append((location, name))
+        # Ignore duplicated entries caused by duplicated paths in `sys.path`.
+        pipeline_entry_points = set(entry_points.get("scancodeio_pipelines"))
 
-    def is_valid(self, pipeline):
+        for entry_point in pipeline_entry_points:
+            pipeline_class = entry_point.load()
+            self.register_pipeline(entry_point.name, pipeline_class)
```

Comment:
Are we are saying here that the name given in the entry IS the name that we will use throughout? eg. as in https://github.com/nexB/scancode.io/pull/95/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R58 ?

I wonder if we should instead may be consider this name to be an internal "key" and instead rely on a name provided as a class attribute instead? ... e.g. we want that key unique but would not use that in the UI directly?